iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_HPP
+
+#include "iceoryx_hoofs/internal/posix_wrapper/mutex.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/subscriber.hpp"
+
+#include "owl/types.hpp"
+
+namespace owl
+{
+namespace kom
+{
+/// @note Once a receive callback has been set, calling the following methods is not thread-safe:
+///           - Subscribe()
+///           - Unsubscribe()
+///           - TakeNewSamples()
+template <typename T>
+class EventSubscriber
+{
+  public:
+    using SampleType = T;
+
+    static constexpr uint64_t HISTORY_REQUEST{1U};
+    static constexpr bool NOT_OFFERED_ON_CREATE{false};
+
+    EventSubscriber(const ServiceIdentifier& service,
+                    const InstanceIdentifier& instance,
+                    const EventIdentifier& event) noexcept;
+
+    /// @note Will disable the receive callback if active
+    void Subscribe(std::size_t queueCapacity) noexcept;
```

Comment:
Add const? Do we really want to use types without a fixed size? I would recommend a fixed size type like `uint64_t` since this is the best practice according to the core guidelines and languages like rust even do not offer non fixed size types.
Those non fixed size types can be the reason that various applications have problems running platform independent.

Reply:
True, one philosophical item we follow in iceoryx is to be as explicit as possible.